docs/proposals/scheduling/20250611-networktopology-aware-scheduling.md

```diff
@@ -0,0 +1,438 @@
+3. `Accelerator Interconnect` is an ultra-high-speed network used to connect multiple GPUs within a computing system, enabling them to communicate at extremely high bandwidth. For example, Nvidia's NVLink, AMD's Infinity Fabric/UALink, Google TPU's ICI, and Amazon Trainium 2's NeuronLink.
+4. `Out of Band Management Networking` is a dedicated network used for system management that is separate from the main data transmission network. This network enables administrators to perform operating system re-imaging, monitor node health (such as fan speed, temperature, power consumption, etc.), and control IT equipment.
+
+When we talk about network topology, most of the time we are talking about Backend Networking and Accelerator Interconnection that connect GPUs. NVIDIA provides a [topograph](https://github.com/NVIDIA/topograph/blob/main/docs/k8s.md) tool to label the nodes in the cluster with their positions in the cluster network topology:
+
+- `accelerator`: Network interconnect for direct accelerator communication (e.g., Multi-node NVLink interconnect between NVIDIA GPUs)
+- `block`: Rack-level switches connecting hosts in one or more racks as a block.
+- `spine`: Spine-level switches connecting multiple blocks inside a datacenter.
+- `datacenter`: Zonal switches connecting multiple datacenters inside an availability zone.
+
+For example, if a node belongs to NVLink domain `nvl1` and connects to switch `s1`, which connects to switch `s2`, and then to switch `s3`, Topograph will apply the following labels to the node:
+
+```yaml
+network.topology.nvidia.com/accelerator: nvl1
+network.topology.nvidia.com/block: s1
+network.topology.nvidia.com/spine: s2
+network.topology.nvidia.com/datacenter: s3
+```
+
+Network topology domains can be organized in a tree structure. For example, the following figure shows a typical spine leaf network architecture. 
+
+![image](/docs/images/networktopo-1.png)
+
+When GPUs on different nodes communicate with each other,
+
+- The fewer hops there are during communication, the lower the communication delay.
+- The more hops there are during communication, the higher the communication delay and the greater the likelihood of congestion on the core switch.
+
+### Parallel Strategies
+
+![image](/docs/images/networktopo-2-dp-and-pp.png)
+
+The above figure shows a PytorchJob training task:
+
+- Create a total of 12 Pods (PP * DP=4 * 3).
+- Each Pod requests all 8 GPU cards on the node.
+
+requirements:
+- The communication between the 8 cards in one Pod is conducted through nvlink. No need for scheduler attention.
+- The communication between pods requires RDMA high-speed network for communication, and the scheduler needs to schedule DP * PP pods into a set of high-performance communication domains.
+
+PodGroup schedules according to the following strategy:
+- If all allocatable nodes satisfy strategy 1, bound podgroup's pod to the nodes.
+- If strategy 1 is not satisfy, try strategy 2, strategy 3, etc. and so on.
```

Comment:
So naturally, there is an enhancement to guarantee the strategy above level-X.
e.g. Restrict tasks under the same leaf.